extensions/tiled-assets/README.md

```diff
@@ -0,0 +1,127 @@
+| abstract            | string                                                        | Brief narrative description of this tile matrix, normally available for display to a human                                                                        |
+| keywords            | [string]                                                      | Unordered list of one or more commonly used or formalized word(s) or phrase(s) used to describe this dataset                                                      |
+| identifier          | string                                                        | **REQUIRED.** Tile matrix identifier. This identifier can be used in templates to replace `{TileMatrix}` parameters.                                              |
+| scaleDenominator    | number                                                        | **REQUIRED.** Scale denominator level of this tile matrix.                                                                                                        |
+| topLeftCorner       | [number]                                                      | **REQUIRED.** Position in CRS coordinates of the top-left corner of this tile matrix.                                                                             |
+| tileWidth           | number                                                        | **REQUIRED.** Width of each tile of this tile matrix in pixels. Pixel buffers are not reflected in this number.                                                   |
+| tileHeight          | number                                                        | **REQUIRED.** Height of each tile of this tile matrix in pixels. Pixel buffers are not reflected in this number.                                                  |
+| matrixWidth         | number                                                        | **REQUIRED.** The number of columns in this tile matrix. Any number between zero and `matrixWidth - 1` can be used to replace the `{TileCol}` template parameter. |
+| matrixHeight        | number                                                        | **REQUIRED.** The number of rows in this tile matrix. Any number between zero and `matrixHeight - 1` can be used to replace the `{TileRow}` template parameter.   |
+| variableMatrixWidth | [Variable Matrix Width Object](#variable-matrix-width-object) | Describes the rows that has variable matix width                                                                                                                  |
+| pixelBuffer         | [Pixel Buffer Object](#pixel-buffer-object)                   | An optional default pixel buffer description object, overriding the default of the pyramid. By default, no pixel buffers are used.                                |
```

Comment:
Do you really need to know the pixel buffer for downloading the assets?
Similar with many of the fields: What is needed to just find and get the data? Could we reference external metadata (e.g. a static TMS metadata document or so) and only provide the essential fields here?

Reply:
The pixel buffer is necessary, as it can actually displace the data, for that amount of pixels. It is optional anyhow, so I would really love to have it in the spec, but if you say it has nothing to do in that, we can still remove it.

Reply:
STACs primary focus is search and discovery. Are pixel buffers useful for that (i.e. displaying it on a web map)? I'm not saying it must be removed, maybe it helps to clarify in the docs what it is useful for and why it's part of STAC with regard to search and discovery.

Reply:
Hi @m-mohr!
Well the reasoning behind storing tiles with overlaps/buffers is processing performance in certain use cases. There are some tile-based processes which require buffers around the processing tiles (e.g. moving window filters such as hillshade) otherwise artifacts are introduced at tile borders.
Now if such a process uses the same tiling scheme as the input dataset and the input dataset does not have tiles with buffers this means that 9 input tiles have to be read (the one in the center matching with the process tile and the 8 neighbours around) but if the tiles are already stored with buffers, the process only has to read one tile.
I agree, for only downloading the assets having to know the pixelbuffer is not essential. It could however introduce confusion if one wanted to determine the exact bounds of a tile using a given tile index and not the metadata stored in the GTiff. If there is no knowledge of pixelbuffers, the bounds would be off by the pixelbuffer width and height.
I would also argue that in many cases the assets are probably used directly as a input of a processing tool and in that case pixelbuffers are an optimized data structure for the beforementioned use cases.

Reply:
Thank you, @ungarj. I fully understand your point and I often suffer the same thing with STAC. We also use it for processing tasks and need to store related data in STAC, but it's not meant to store such data as the focus was defined to be search and discovery.
@cholmes How to proceed in such cases?

Reply:
I think I've been consistent on my take in cases when people want to put in non-search metadata - STAC is focused on search and discovery, but we provide extension mechanisms for people to add in what they want. So I'm not opposed to extensions that do more of that type of stuff. 

The eventual line I think will be that 'common metadata' should just be things for search. But there can be a rich ecosystem of extensions that people use for more than the core. Since this is just a proposed extension for now I think it's fine. If it evolves to be part of common metadata then it could make sense to remove it when we fold that in - but I think there are quite a few steps to get there.